frontend/src/pages/NewRunV2.tsx

```diff
@@ -288,11 +301,13 @@ function NewRunV2(props: NewRunV2Props) {
       description: runDescription,
       display_name: runName,
       experiment_id: apiExperiment?.id,
-      // pipeline_spec and pipeline_version_id is exclusive.
-      pipeline_spec: !(existingRun?.pipeline_version_id || existingPipelineVersion?.id)
+      // pipeline_spec and pipeline_version_reference is exclusive.
+      pipeline_spec: !(pipelineVersionRefClone || pipelineVersionRefNew)
```

Comment:
Correct me if I am wrong: `pipelineVersionRefClone` is an object while `pipeline_version_id` is a string. And `{}` is truly instead of falsy: https://stackoverflow.com/questions/21743921/javascript-is-an-empty-object-a-falsy-one.

Reply:
Thanks for the reminder! If we get a run / recurring created by `pipeline_spec`, the `pipeline_version_reference` in the API return value is undefined rather than `{}`.

Reply:
Sounds good, thank you for confirming Joe!